nix/default.nix

```diff
@@ -0,0 +1,64 @@
+{
+  pkgs ? import <nixpkgs> { },
+
+  ## REVIEW: Probably hard to provide without a flake
+  advisory-db,
+  crane,
+  rust-overlay,
+}:
```

Comment:
Are we fine with this interface? I put the default `import <nixpkgs> {}` as a reflex for `pkgs`, but it is a bit useless if the three others are so hard to provide.

We could also look into fancy solutions where we `fetchgit` those dependencies, maybe just grabbing their main branch, or maybe grabbing by default the revision from `flake.lock`. That sounds a bit heavy to me, though, but it would allow people to `nix-shell` without having to pin an entire `nixpkgs`.